Create chat view

root8000 opened the pull request:
# WHAT
Create chat view
# WHY
必須機能のため

Commit:
create_Messages_Controller

メッセージコントローラを作成

Commit:
set_gem(haml-rails,erb2haml)

Gem(haml-railsとerb2haml)を導入

Commit:
set_provisional_root

仮置きのルーティングとして、messagesコントローラのindexアクションがrootパスになるよう設定

Commit:
create_index.html

indexビューファイルの作成

Commit:
replace_.erb_for_.haml

既存のerbファイルを全てhamlに置き換え

Commit:
set_index_action_at_messages_controller

indexアクションをmessagesコントローラに定義

Commit:
rename_index.html_to_index.html.haml

index.htmlをindex.html.hamlにファイル名変更

Commit:
recreate_messages_controller

不要ファイル削除のため、一度messagesコントローラを削除。不要ファイルが作成されないようconfig/apprication.rbに追記し、再度messagesコントローラを作成。

Commit:
destroy_unnecessary_file

config/apprication.rbへの不要ファイルへの追記を、コントローラ削除の前にしてしまったため、不要ファイルが削除されていなかった。追記を一度外し、再度コントローラ削除。その後追記したのちコントローラを再作成。無事不要ファイルをのぞいてmessagesコントローラが作成された。

Commit:
comment_out_messages_routing(get_messages/index)

messagesコントローラ作成時に自動で作られたルーティング(get 'messages/index')を一時的にコメントアウト

Commit:
edit_index.html.haml

index.html.hamlの雛形を作成

Commit:
modify_index.html.haml

index.html.hamlファイル内のクラス名を変更

Commit:
add font-awesome rails

faアイコン利用のためfont-awesome-railsを導入

Commit:
modify index.html.haml

全体的に命名を変更。正解がわからない。

Commit:
create _main_content_messages.html.haml

今後の機能を見据えて、main-contentのメッセージ一覧を部分テンプレート化

Commit:
create _left_content_groups.html.haml

今後の機能を見据えて、left-contentのグループ一覧を部分テンプレート化

Commit:
modify messages.scss

messagesのscssを全体的に編集。Sassよくわからない。

Commit:
destroy application.css

scssファイルに変更しようとしたが名称変更してもcssのままのようだったので、元のapplication.cssは削除

Commit:
create application.css.scss

cssファイルのscssへの名称変更がうまくいかなかったため、新たにapplication.css.scssを作成

Commit:
create _variable.scss

_variable.scssを作成。使い方がよくわからない。

Commit:
create _reset.scss

_reset.scssを作成

taketnk commented on app/assets/stylesheets/_variable.scss:
```diff
@@ -0,0 +1,24 @@
+    clear: both;
+  }
+}
+$mainWhiteColor: #FFFFFF;
```
variable.scssはその名の通り、変数を定義するファイルにしたいので、8 ~ 10行目以外は別ファイルに記述しておいてください。

root8000 replied:
それぞれ別ファイルに記述しました。

taketnk commented on app/assets/stylesheets/application.css.scss:
```diff
@@ -0,0 +1,4 @@
+@import "./reset";
+@import "./messages";
+@import "./variable";
```
messagesよりも先にvarialeを読み込ませておかないと、messagesのファイルでは、変数が利用できませんね。

root8000 replied:
ありがとうございます！！！どうしてvariableだけ両ファイルに書かないと動かないのか凄まじく謎だったのですが、順番が大事だったのですね。書き順を修正しました。

taketnk commented on app/assets/stylesheets/messages.scss:
```diff
@@ -0,0 +1,175 @@
+@import "./variable";
```
上記を書き換えれば、こちらの行は不要になります。

root8000 replied:
削除しました。

taketnk commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,35 @@
+    .left-content__header
+      .left-content__header__user-name
+        ユーザー名
+      %a{class: "left-content__header__new-group", href: "/groups/new"}
```
link_toでパスを作成しましょう。

root8000 replied:
link_toに書き換えました。

taketnk commented on config/application.rb:
```diff
@@ -11,5 +11,10 @@ class Application < Rails::Application
     # -- all .rb files in that directory are automatically loaded.
+    config.generators do |g|
+      g.javascripts false
+      g.helper false
```
いいですね〜

root8000 replied:
ありがとうございます！！

Commit:
change a-tag for link_to

aタグの記述をlink_toへ変更

Commit:
create mixin folder and _crearfix.scss

valiable.scssに記述してしまっていた@mixin clearfixを、mixinフォルダを作り、そこに格納

Commit:
detach import variable

application.css.scssの編集により、messages.scssにvariableの記述が必要なくなったため削除

Commit:
modify application.css.scss

messeagesより後にvariableを記述していたために、messagesでvariableを読み込めていなかった。書き順の修正。
また、新たに作成したmixinファイルのcrearfix.scssを読み込むよう追記

Commit:
modify variable.scss

変数以外の記述を削除

MrUrakou reviewed (commented):
すみません、確認のためviewの表示画面のスクショを添付していただくと助かります。

MrUrakou commented on app/views/layouts/application.html.haml:
```diff
@@ -0,0 +1,10 @@
+!!!
+%html
+  %head
+    %meta{:content => "text/html; charset=UTF-8", "http-equiv" => "Content-Type"}/
```
ハッシュロケットでない書き方に変えましょう。

root8000 replied:
変更しました。

MrUrakou commented on app/views/layouts/mailer.html.haml:
```diff
@@ -0,0 +1,8 @@
+!!!
+%html
+  %head
+    %meta{:content => "text/html; charset=utf-8", "http-equiv" => "Content-Type"}/
```
ハッシュロケットでない書き方にしましょう。

root8000 replied:
変更しました。

MrUrakou commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,35 @@
+          %i グループメンバー
+    .main-content__body
+      .main-content__body__messages-list
+        =render partial: "main_content_messages"
```
細かいですが、=のあとは半角スペース空けましょう。

root8000 replied:
失礼しました！半角スペース空けました。

MrUrakou commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,35 @@
+    .left-content__header
+      .left-content__header__user-name
+        ユーザー名
+      =link_to "/groups/new", class: "left-content__header__new-group" do
```
パスを指定する際には直接URLを指定するのではなくて、名前付きのヘルパーで指定しましょう。以下同様です。

root8000 replied:
まだcomment以外のモデルやコントローラの作成をしていないので、ちょっと合っているか怪しいのですが、仮置きで記述してみました。" "で囲んでおかないとそんなページやらメソッドやらないよ！と出てしまうので、いったん囲っています。

MrUrakou commented on app/views/messages/index.html.haml:
```diff
@@ -0,0 +1,35 @@
+        ユーザー名
+      =link_to "/groups/new", class: "left-content__header__new-group" do
+        %i.fa.fa-pencil-square-o
+      =link_to "/user/edit", class: "left-content__header__edit-user" do
```
パスを指定する際には直接URLを指定するのではなくて、名前付きのヘルパーで指定しましょう。

Commit:
set a-tag scss

記述漏れしていたaタグについて追記

Commit:
replace hash rocket by symbol

ハッシュロケットによる記述をシンボルへ変更

Commit:
add link_to to _left-content_groups.html.haml

各グループへのリンクを部分テンプレートに持ってき忘れていたので追記

Commit:
replace url by path

link_toのurlをパス表記に変更。ルーティング設定がまだなので、パス指定や引数やらが合っているかだいぶ怪しい。ルーティング設定後に要修正。
